src/assignments/tech-jobs-console.rst

```diff
@@ -0,0 +1,492 @@
+providing access to.
+
+The Data File: ``jobs_data.csv``
+--------------------------------
+
+Our simple app doesn’t connect to a database. If the prototype proves
+useful and we continue development, we’ll add that functionality later.
+But for now, we’ve been given a CSV (comma-separated values) file from
+the Company Team at LaunchCode that contains some recent jobs. This file
+was exported from an Excel spreadsheet into this format, which is easy
+for programs to read in.
+
+If CSV files are new to you, don’t worry, they’re easy to understand.
+CSV files are conceptually similar to simple spreadsheets in that they
+organize data in rows and columns. The major difference is that they
+don’t have the ability to carry out calculations the way spreadsheets
+do, and you can easily open, read, and edit them in plain text editors.
+
+Open up ``jobs_data.csv``. You’ll see that the first line is:
+
+.. sourcecode:: bash
+
+   name,employer,location,position type,core competency
+
+While it isn’t required, the first line of a CSV file often represents
+the column names. We have 5 names here, which indicates that each of our
+rows in the CSV file should have 5 fields. In this file format, a *row*
+corresponds to a new line. So each line below the first will constitute
+a row of data, or a record.
+
+Have a look at the data below line 1, and ask yourself the following
+questions:
+
+#. Which fields match up with which column names above?
+#. Why do some lines/rows (e.g. line 10) have more commas than others, if
+   commas are supposed to separate columns?
+#. What role do the double-quotes play?
+
+The ``Program`` Class
+---------------------
+
+The ``Program`` class contains the ``Main`` method that will drive our
+program’s functionality. It contains three methods:
+
+#. ``Main`` - The main application runner.
+#. ``GetUserSelection`` - A utility method that displays a menu of choices and
+   returns the user’s selection.
+#. ``PrintJobs`` - This is meant to print a list of jobs to the console in a
+   nicely formatted manner, but hasn’t been implemented yet. This will be part
+   of your job.
+
+Let’s look at each of these.
+
+The ``Main`` Method
+^^^^^^^^^^^^^^^^^^^
+
+The logic within ``Main`` presents menus in turn and based on the
+user’s choice, takes appropriate action.
+
+It begins by declaring two local variables: ``columnChoices`` and
+``actionChoices``. These contain information relating to the menus that
+we’ll display, and we’ll look at them in more detail later.
+
+Next, we notice a ``while loop`` that starts ``while (true)``. While we usually
+want to avoid creating infinite loops, we have a good reason for doing so in
+this case! We want our application to continually run until the user decides
+they want to quit. The simplest way to do this is to loop forever. When the
+user wants to quit, they can kill our program by pressing *ctrl+C* (a
+widely-known command to kill a console application). 
```

Comment:
i'm not sure i see how a do-while would benefit this solution or provide something substantially different? doesnt it still create the same situation?

Reply:
It would! I can just see students asking why the starter code doesn't use one! If you are not worried about it, then no need to add any additional explanation.